src/metabase/lib/join.cljc

```diff
@@ -271,6 +277,117 @@
   [x fields]
   (with-join-fields-method x fields))
 
+(defn- select-home-column
+  [home-cols cond-fields]
+  (let [cond->home (into {}
+                         (keep  (fn [home-col]
+                                  (when-let [cond-field (lib.equality/find-closest-matching-ref
+                                                         (lib.ref/ref home-col)
+                                                         cond-fields)]
+                                    [cond-field home-col])))
+                         home-cols)
+        cond-home-cols (map cond->home cond-fields)]
+    ;; first choice: the leftmost FK or PK in the condition referring to a home column
+    (or (m/find-first (some-fn lib.types.isa/foreign-key? lib.types.isa/primary-key?) cond-home-cols)
+        ;; otherwise the leftmost home column in the condition
+        (first cond-home-cols)
+        ;; otherwise the first FK home column
+        (m/find-first lib.types.isa/foreign-key? home-cols)
+        ;; otherwise the first PK home column
+        (m/find-first lib.types.isa/primary-key? home-cols)
+        ;; otherwise the first home column
+        (first home-cols))))
+
+(defn- strip-id [s]
+  (when (string? s)
+    (str/trim (str/replace s #"(?i) id$" ""))))
+
+(defn- similar-names [name0 name1]
```

Comment:
I'm sure this does something important but I am really having a tough time understanding what it does just based on the name. Is this a predicate function... in which case should it be called `similar-names?`. Even tho this is private and we don't require a docstring, a short docstring would go a long way in making this code easier to grok

Reply:
Added the question mark and a comment. (BTW, this is basically ported from the FE code.)